### DISCUSSION TITLE
scheme / host variable in file storage link

### DISCUSSION THREAD

**Discussion by ghost** [btcpayserver/btcpayserver#2333] [Ideas / Feature Requests]
My concern is that if I put a clearnet link to a custom css from the local file server in the App settings, but the user reaches the btcpay PoS through Tor, that the custom css will be downloaded through an exit node. 

Is it possible to set a variable in the link for the internal file storage? 

`$scheme://$host/`, where `$scheme` is either `http` or `https`, and `$host` is either `domain.com` or `foo.onion`. 

This way, if the user talks to the clearnet server, he gets the file through the clearnet domain, but if he talks to the onion service, he will stay inside Tor and not use an exit node.

**Comment by Kukks** [comment]
Makes sense. I think  I would do it that you can instead input the file id in the image/css/file links and it gets resolved on the server behind the scenes. 

We can further improve this later with a File picker that shows all uploaded files

**Reply by ghost** [reply]
yes, that's a nice idea, and solves the issue I'm dealing with.

Thanks! :black_heart:
